### PR TITLE
specs and env options

### PR DESCRIPTION
**Pull request by suryart**
This PR adds:

- --spec/--specs/-s option for running specific specs specified in option on BrowserStack
- --env option for setting specified option on BrowserStack, it works as same as [cypress' --env option](https://docs.cypress.io/guides/guides/environment-variables.html#Option-4-env), hence we do not have nested support as well.

### CODE REVIEW

File: bin/helpers/utils.js [hunk 1]
```diff
@@ -95,6 +95,32 @@ exports.setBuildName = (bsConfig, args) => {
...
 }
 
+// specs can be passed from bstack configuration file
+// specs can be passed via command line args as a string
+// command line args takes precedence over config
+exports.setUserSpecs = (bsConfig, args) => {
+  let bsConfigSpecs = bsConfig.run_settings.specs;
+
+  if (!this.isUndefined(args.specs)) {
+    bsConfig.run_settings.specs = args.specs.split(/\s{0,},\s+/).join(',');
```
Comment (nagpalkaran95) [inline bin/helpers/utils.js:105]: Check for array as well just like `bsConfig.run_settings.specs`

File: bin/helpers/utils.js [hunk 2]
```diff
@@ -95,6 +95,32 @@ exports.setBuildName = (bsConfig, args) => {
...
+    bsConfig.run_settings.specs = bsConfigSpecs.split(/\s{0,},\s+/).join(',');
+  } else {
+    bsConfig.run_settings.specs = null;
+  }
+}
+
+// env option must be set only from command line args as a string
+exports.setTestEnvs = (bsConfig, args) => {
+  if (!this.isUndefined(args.env)) {
+    bsConfig.run_settings.env = args.env.split(/\s{0,},\s+/).join(',');
```
Comment (nagpalkaran95) [inline bin/helpers/utils.js:118]: `.split(/\s{0,},\s+/).join(',')` move this to a func? being used 3 time in the last last lines of code

File: bin/runner.js [hunk 3]
```diff
@@ -164,6 +164,17 @@ var argv = yargs
           describe: Constants.cliMessages.RUN.BUILD_NAME,
           type: "string",
           default: undefined
+        },
+        's': {
+          alias: ['specs', 'spec'],
+          describe: Constants.cliMessages.RUN.SPECS_DESCRIPTION,
+          type: "string",
+          default: undefined
```
Comment (nagpalkaran95) [inline bin/runner.js:172]: confirm the cli messages and default values once

File: test/unit/bin/helpers/capabilityHelper.js [hunk 4]
```diff
@@ -249,6 +249,136 @@ describe("capabilityHelper.js", () => {
           chai.assert.fail("Promise error");
         });
     });
+
+    context("specs and env from run_setting", () => {
+      it("sets specs list is present", () => {
```
Comment (nagpalkaran95) [inline test/unit/bin/helpers/capabilityHelper.js:254]: no test where specs list is an Array or passed args is of Array type

**Review by nagpalkaran95 — CHANGES_REQUESTED** [review]

File: test/unit/bin/helpers/capabilityHelper.js (hunk 4 above)
Comment (suryart) [reply test/unit/bin/helpers/capabilityHelper.js:254]: As call to `util.setUserSpecs` and `util.setTestEnvs`are [made in runs.js](https://github.com/browserstack/browserstack-cypress-cli/pull/46/files/961eb70b7347802f7cad51a6192c92b6ec21df39#diff-94102bcdbd99f5dee9c6696fdae0a977R27-R32) and then final state(String) is passed into capabilityHelper. Hence, capabilityHelper always receives string.  

specs for array conditions are covered in the util.js itself: https://github.com/browserstack/browserstack-cypress-cli/pull/46/files/961eb70b7347802f7cad51a6192c92b6ec21df39#diff-175b2b2ef3763bf0f62fe44137aa5d34R350

**Review by nagpalkaran95 — APPROVED** [review]
lgtm

**Comment by prateekbstack** [comment]
This PR is good to go. Sorry for the delay in response.

**Review by nagpalkaran95 — APPROVED** [review]

**Pull request merged**